test/e2e/test/watcher.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+type Watcher struct {
```

Comment:
Can we refactor Elasticsearch `checks_budget.go` using this new `Watcher` mechanism? (I'm fine with doing that in another PR or just opening an issue).

Reply:
Would the continuous health check in `steps_mutation.go` be another candidate for this?

Reply:
I hoped to do it in a different PR, filed https://github.com/elastic/cloud-on-k8s/issues/2174 to track.